### ISSUE TITLE
make v2 the default branch

### ISSUE THREAD

**Issue by natefinch** [volatiletech/authboss#179]
It's very confusing that the default view of the code is the deprecated version.  There's an option in github to make a branch the default view (rather than master).  Could you do that, so that when people come here, they don't get confused?

**Comment by natefinch** [comment]
(goes for the sample, too... )

**Comment by aarondl** [comment]
I agree with the sentiment wholeheartedly. I'm hoping to release v2 shortly though and there's still been some breaking changes in the rc happening including one coming soon as I'm adding otp and 2fa support to Authboss, and I know that shouldn't happen, last minute features :(. Changing the default branch would change what people get when they `go get` which would leave them susceptible to breaking changes on subsequent `go get`s when their expectation of a master branch should be stability and compatibility (in the pre-vgo/dep world).

I'm all in favor of pushing my agenda of promoting v2 but I don't want to cause harm to developers who chose to use the library. If there's some way to do this without potentially breaking people I'm keen to, if there's solutions I'm not thinking of by all means let me know. Otherwise I'd ask that we just sit tight for just a little while longer. Thoughts?

**Comment by natefinch** [comment]
FYI, changing the default branch does not change what people get when they go-get.  They'll get master no matter what. Default branch is purely a github concept for what is shown on the homepage of your url.  (I have projects that use master and v2 projects on github).

I'm developing against AB v2 right now, (by telling dep to use `branch = v2` in its configuration).

**Comment by aarondl** [comment]
I don't believe this is true.

I found one of your repositories that's got a master branch and a default branch that is not master (lumberjack).

Plain git obeys the default branch set on github.

```bash
$ git clone git@github.com:natefinch/lumberjack.git
$ git rev-parse --abbrev-ref HEAD
heads/v2.0
$ git lg -2
* aee4629 - (HEAD -> v2.0, origin/v2.0, origin/HEAD) Update docs, adding `Compress` setting details (#49) (11 months ago)
* df99d62 - switch to travis (#44) (1 year, 2 months ago)
$ git bl
* v2.0                        aee4629 [origin/v2.0] Update docs, adding `Compress` setting details (#49)
  remotes/origin/HEAD         -> origin/v2.0
  remotes/origin/master       6d54cbc Add deprecation notice
  remotes/origin/v2.0         aee4629 Update docs, adding `Compress` setting details (#49)
  ... elided irrelevant extras ...
```

```bash
# Ensured everything was deleted first, so the -u is pointless :p
$ go get -u github.com/natefinch/lumberjack
$ git bl
* v2.0                        aee4629 [origin/v2.0] Update docs, adding `Compress` setting details (#49)
  remotes/origin/HEAD         -> origin/v2.0
  remotes/origin/master       6d54cbc Add deprecation notice
  remotes/origin/v2.0         aee4629 Update docs, adding `Compress` setting details (#49)
  ... elided irrelevant extras ...
```

This simple test validates what I was saying. But maybe this repository has been changed in some way I'm not aware of to make it so it pulls v2? If so let me know that trick so I can force it to pull master, and change my default branch and that'd solve this problem :)

**Comment by natefinch** [comment]
Wow uh, yeah, I just tried that and I am quite surprised.  I guess you learn something every day :)   Nevermind then :)

**Comment by fjorgemota** [comment]
What about using gopkg.in to help on the versioning?

Instead of using the direct link of the repository, you could use `http://gopkg.in/volatiletech/authboss.v1` for the version 1, and `http://gopkg.in/volatiletech/authboss.v2` for the version 2. That way, even the docs would work nicely for both versions: You can check the Godoc for [version 1](http://godoc.org/gopkg.in/volatiletech/authboss.v1) and for [version 2](https://godoc.org/gopkg.in/volatiletech/authboss.v2), for example, without any additional tools.

What do you think?

**Comment by aarondl** [comment]
Unfortunately we won't be moving this library back to gopkg.in, the approved solution for dependency management in Go is modules which we'll be switching to as soon as Go 1.11 lands. It's also true that tooling like godoc.org is going to be updated to work well with modules in the future so there will be no advantage to gopkg.in in that sense.

It's also worth pointing out that the main issue reported here is that the master branch being v1 not v2 is the part that's confusing people. And that wouldn't change even if we did have a v1 and v2 branch to be used by gopkg.in since technically v2 is "unstable" and so master would still point to the HEAD of v1. As I advised nate, we'll just have to wait for a little bit until v2 is ready. It shouldn't be too much longer as I'm back from vacation now!
